src/utils.jl

```diff
@@ -0,0 +1,158 @@
+     param X : first string in tokenized form
+    type (X) : Array{SubString{String},1}
+     param Y : second string in tokenized form
+    type (Y) : Array{SubString{String},1}
+     param weighted : Weighted LCS is done if weighted is True (default)
+    type (weighted) : Boolean
+     param return_string : Function returns weighted LCS length when set to False (default).
+                           Function returns longest common substring when set to True.
+    type (return_string) : Boolean
+     param f: weighting function. The weighting function f must have the property
+              that f(x+y) > f(x) + f(y) for any positive integers x and y. 
+    type (f) : generic function which takes a float as an input and returns a float.
+    =#
+    
+    m, n = length(X), length(Y)
+    c_table = [zeros(n+1) for i in 1:m+1]
+    w_table = [zeros(n+1) for i in 1:m+1]
+    
+    for i in 1:(m+1)
+        
+        for j in 1:(n+1)
+            
+            if i == 1 || j == 1
+                continue
+            
+            elseif X[i-1] == Y[j-1]
+                
+                k = w_table[i-1][j-1]
+                if weighted == true
+                    increment = (f(k+1)) - (f(k)) 
+                else
+                    increment = 1
+                end
+                c_table[i][j] = c_table[i-1][j-1] + increment
+                w_table[i][j] = k + 1
+                
+            else
+                
+                if c_table[i-1][j] > c_table[i][j-1]
+                    c_table[i][j] = c_table[i-1][j]
+                    w_table[i][j] = 0  # no match at i,j
+                else
+                    c_table[i][j] = c_table[i][j-1]
+                    w_table[i][j] = 0  # no match at i,j
+                end
+            
+            end
+        
+        end
+    
+    end
+    
+    lcs_length = (c_table[m+1][n+1])
+    if return_string == false
+        return lcs_length
+    end
+    
+    if weighted == true
+        lcs_length = c_table[m][n]^(2)
+    end
+    
+    lcs_length = round(Int64, lcs_length)
+    lcs_length = convert(Int64, lcs_length)
+    lcs = ["" for i in  1:(lcs_length+1)]
+    lcs[lcs_length+1] = ""
+    i = m+1
+    j = n+1
+    
+    while i>1 && j>1
+        if X[i-1] == Y[j-1]
+            lcs[lcs_length+1] = X[i-1]
+            i -= 1
+            j -= 1
+            lcs_length -= 1
+
+        elseif c_table[i-1][j] > c_table[i][j-1]
+            i -= 1
+        else
+            j -= 1
+        end
+    end
+    
+    return (join(lcs, " "))  # the lcs string
+
+end
+
+function FMeasureLCS(RLCS, PLCS, beta=1)
+    #=F-measure based on WLCS
+    
+    param beta : user defined parameter
+    type (beta) : float
+    
+    param r_lcs : recall factor
+    type (r_lcs) : float
+    
+    param p_lcs : precision factor
+    type (p_lcs) : float
+     
+    score : f measure score between a candidate
+    	    and a reference
+    =#
+    
+    try
```

Comment:
The try-catch is slow and can be replaced for faster performance, the statement can be checked for a `DivideError` beforehand.

Reply:
Wouldn't make much of a difference as it is a one-time calculation. 
Also, it would allow me to rethrow the error in case it is not a DivideError
If there is a method that allows me to catch a Divide Error while rethrowing any other error let me know.

Reply:
It is better to avoid errors in a package rather than relying on try-catch to do it for. You may go through the following doc for more details. 

https://docs.julialang.org/en/v1/manual/style-guide/index.html#Don't-overuse-try-catch-1

Reply:
There is an extensive debate on try-catch which I don't want to enter. 
Also, Julia docs are very poorly maintained. Referring to a sentence does very little to convince me of the backlash of using try-catch. 
Also, try-catch-rethrow is for user errors and not for package errors. 
All good packages in other languages make sure to use try-except for exception handling for every function. There wouldn't be an extensive subject topic on the matter if you could refer to a single line in the doc and dismiss it.